joomla/joomla-cms
# Downloadable system information

This replaces https://github.com/joomla/joomla-cms/pull/5647
## Description

This adds a new button in backend: `System` > `System Information` that allows administrator to download the page settings. This can be useful to debug system issues by support teams.

![download-system-information](https://cloud.githubusercontent.com/assets/1119272/6642766/9ad5ec00-c9a7-11e4-8ce4-a5e916238a0f.png)

This also adds a new fast method to add a backend toolbar button through: `JToolbarHelper::link($url, $text, $name = 'link')` which will be useful to allow external links on components toolbars.
## How to test it
1. Go to system information and see that all the settings are still there as expected
2. Click on download and make sure that all the sections are there. It contains:
3. `php_settings`: Information about the php settings 
4. `config`: Configuration information
5. `info`: System information
6. `php_info`: Parsed `phpinfo()` information
7. `directories`: Information about the website folders (if they are writable or not)

I have changed the original way that @Radek-Suski used to parse the file as I think it should be processable by third part systems. So the most real the data is the better.

Private / Critical information has to NOT be exported in any way so if you detect any information that should be removed report it.
## Backward compatibility issues

This is a new feature and shouldn't create any B/C issue. The only method affected is `AdminModelSysInfo::getPhpInfo()` that has been renamed and optimized.
